opensearch/examples/json.rs

```diff
@@ -0,0 +1,136 @@
+use serde_json::{json, Value};
+
+use opensearch::auth::Credentials;
+use opensearch::cert::CertificateValidation;
+use opensearch::http::headers::HeaderMap;
+use opensearch::http::transport::{SingleNodeConnectionPool, TransportBuilder};
+use opensearch::http::{Method, Url};
+use opensearch::OpenSearch;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let url = Url::parse("https://localhost:9200")?;
+    let credentials = Credentials::Basic("admin".into(), "admin".into());
+    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
+        .cert_validation(CertificateValidation::None)
+        .auth(credentials)
+        .build()?;
+    let client = OpenSearch::new(transport);
+
+    let index_name = "movies";
+    let document_id = "1";
+
+    let info: Value = client
+        .send(
+            Method::Get,
+            "/",
+            HeaderMap::new(),
+            Option::<&String>::None,
+            Option::<&String>::None,
+            None,
+        )
+        .await?
+        .json()
+        .await?;
+
+    println!(
+        "Welcome to {} {}",
+        info["version"]["distribution"], info["version"]["number"]
+    );
+
+    // Create an index
+    let index_body = json!({
+        "settings": {
+            "index": {
+                "number_of_shards" : 4
+            }
+        }
+    });
+
+    let create_index_response = client
+        .send(
+            Method::Put,
+            &format!("/{index_name}"),
+            HeaderMap::new(),
+            Option::<&String>::None,
+            Some(index_body.to_string()),
+            None,
+        )
+        .await?;
+
+    assert_eq!(create_index_response.status_code(), 200);
+
+    // add a document to the index
+    let document = json!({
+        "title": "Moneyball",
+        "director": "Bennett Miller",
+        "year": "2011"
+    });
+    let create_document_response = client
+        .send(
+            Method::Put,
+            &format!("/{index_name}/_doc/{document_id}"),
+            HeaderMap::new(),
+            Some(&vec![("refresh", "true")]),
```

Comment:
I'm thinking of opening an issue to specialize the type for the http query in this particular method, as [under the hood](https://github.com/seanmonstar/reqwest/blob/master/src/async_impl/request.rs#L354), it uses [serde url-encoded](https://github.com/nox/serde_urlencoded). 

Since the only constraint on the [Query type](https://github.com/opensearch-project/opensearch-rs/blob/main/opensearch/src/client.rs#L100) `Q` is `Serialize`, it's easy to think any `Serialize` type works, but not everyone works, and the sad thing is that the error is not caught at Compile type, but at Runtime with [this error](https://github.com/nox/serde_urlencoded/blob/d8bc15d16adf6b3ba6ae46d199d7109cf5079efa/src/ser/value.rs#L60)
I'm thinking of specializing the [query string](https://github.com/opensearch-project/opensearch-rs/blob/main/opensearch/src/client.rs#L94) to 
```
query_string: Option<&[(&String, &String)]>,
```

WDYT ?